components/notification-generator/minecraftTextParser.tsx

```
@@ -0,0 +1,117 @@
+      const miniMessageString = convertLegacyToMini(cleanedText);
+      const deserialized = MiniMessage.miniMessage().deserialize(miniMessageString);
+      return MiniMessage.toHTML(deserialized);
+    } catch (error) {
+      console.warn('MiniMessage parse failed, falling back to legacy parser:', error);
+      return null;
+    }
+  }, [cleanedText]);
+
+  if (htmlOutput) {
+    return (
+      <span className="font-minecraft" dangerouslySetInnerHTML={{ __html: htmlOutput }} />
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)